src/haz3lcore/dynamics/Constraint.re

```diff
@@ -119,15 +119,23 @@ let rec ctr_of_nth_variant = (num_variants, nth): (t => t) =>
     xi => InjR(xi |> ctr_of_nth_variant(num_variants - 1, nth - 1));
   };
 
-let of_ap = (ctx, ctr: option(Constructor.t), arg: t): t =>
+let of_ap = (ctx, mode, ctr: option(Constructor.t), arg: t): t =>
   switch (ctr) {
   | Some(name) =>
-    switch (Ctx.lookup_ctr(ctx, name)) {
-    | None => Falsity // TODO: review
-    | Some({num_variants, nth, _}) =>
-      arg |> ctr_of_nth_variant(num_variants, nth)
+    switch (mode) {
+    | Mode.Ana(ty) =>
+      switch (Typ.weak_head_normalize(ctx, ty)) {
+      | Sum(map) =>
+        let num_variants = ConstructorMap.cardinal(map);
+        switch (ConstructorMap.nth(name, map)) {
+        | Some(nth) => arg |> ctr_of_nth_variant(num_variants, nth)
+        | None => Falsity // TODO: review
+        };
+      | _ => Falsity // TODO: review
```

Comment:
This constraint value is never used. It will be replaced with `Hole` after error checkings, since the pattern doesn't have the type of the scrutinee.